ovos_gui_plugin_shell_companion/brightness.py

```diff
@@ -232,36 +298,73 @@ def get_sunset_time(self, message=None):
+    @property
+    def auto_night_mode_enabled(self) -> bool:
+        """
+        Check if auto night mode is enabled.
+
+        Returns:
+            bool: True if auto night mode is enabled, False otherwise.
+        """
+        return self.config.get("auto_nightmode", False)
+
+    def start_auto_night_mode(self):
+        """
+        Start the auto night mode functionality.
+        """
+        LOG.debug("Starting auto night mode")
+        self.config["auto_nightmode"] = True
+        update_config("auto_nightmode", True)
+
+        now = now_local()
+        if self.sunset_time < now < self.sunrise_time:
+            self.handle_sunset()
+        else:
+            self.handle_sunrise()
+
+    def handle_sunrise(self, message: Optional[Message] = None):
+        """
+        Handle the sunrise event for auto night mode.
+
+        Args:
+            message: Optional message received from the bus.
+        """
         if self.auto_night_mode_enabled:
-            date = now_local()
-            self.event_scheduler.schedule_event(self.start_auto_night_mode,
-                                                when=date + timedelta(hours=1),
-                                                name="ovos-shell.night.mode.check")
-            if self.sunset_time < date < self.sunrise_time:
-                LOG.debug("It is night time")
-                self.bus.emit(Message("phal.brightness.control.auto.night.mode.enabled"))
-            else:
-                LOG.debug("It is day time")
-                # TODO - implement this message in shell / check if it exists
-                # i just made it up without checking
-                self.bus.emit(Message("phal.brightness.control.auto.night.mode.disabled"))
+            LOG.debug("It is daytime")
+            self.bus.emit(Message("ovos.homescreen.main_view.current_index.set",
+                                  {"current_index": 1}))
 
-    def stop_auto_night_mode(self):
-        LOG.debug("Stopping auto night mode")
-        self.auto_night_mode_enabled = False
-        self.event_scheduler.cancel_scheduled_event("ovos-shell.night.mode.check")
+            self.event_scheduler.schedule_event(self.handle_sunset,
+                                                when=self.sunset_time,
+                                                name="ovos-shell.sunset")
 
-    def is_auto_night_mode_enabled(self):
-        display_config = get_ovos_shell_config()
-        self.auto_night_mode_enabled = display_config.get("auto_nightmode", False)
+    def handle_sunset(self, message: Optional[Message] = None):
+        """
+        Handle the sunset event for auto night mode.
 
+        Args:
+            message: Optional message received from the bus.
+        """
         if self.auto_night_mode_enabled:
-            self.start_auto_night_mode()
-        else:
-            self.stop_auto_night_mode()
+            LOG.debug("It is nighttime")
+            # show night clock in homescreen
```

Comment:
I haven't been clear about the intended behavior here - is nightmode only supposed to show the night clock, or should it also enable auto_dim?

What makes sense to me is showing the night clock _and_ enabling auto_dim at night, then in the morning showing the default screen and disabling auto_dim until sunset.

Reply:
nightmode and autodim are separate and completely independent settings

nightmode only shows the dark face clock from homescreen, nothing else... personally i dont see the value in that  (i think that screen is ugly and never want to see it :upside_down_face: )

with this PR i only wanted to streamline the logic but have it still doing the same thing, but i def agree we should define what nightmode means and extend it, it should probably do a bunch of more stuff like adjusting volume and max brightness etc. @NeonDaniel want to open an issue and spec this out? you probably have more feedback from mk2 users where this matters most

Reply:
id be ok with just making nightmode enabling/disabling the auto-dim option, that is straightforward enough for an initial implementation and i can do it in this refactoring round